apps/wpcom-block-editor/src/wpcom/features/live-preview.tsx

```diff
@@ -0,0 +1,118 @@
+import domReady from '@wordpress/dom-ready';
+import { __, sprintf } from '@wordpress/i18n';
+import { registerPlugin } from '@wordpress/plugins';
+import { __dangerousOptInToUnstableAPIsOnlyForCoreModules } from '@wordpress/private-apis';
+import { getQueryArg } from '@wordpress/url';
+import { useEffect } from 'react';
+
+/**
+ * Return true if the user is currently previewing a theme.
+ * FIXME: This is copied from Gutenberg; we should be creating a selector for the `core/edit-site` store.
+ * @see https://github.com/WordPress/gutenberg/blob/053c8f733c85d80c891fa308b071b9a18e5194e9/packages/edit-site/src/utils/is-previewing-theme.js#L6
+ * @returns {boolean} isPreviewingTheme
+ */
+function isPreviewingTheme() {
+	return getQueryArg( window.location.href, 'wp_theme_preview' ) !== undefined;
+}
+
+/**
+ * Return the theme slug if the user is currently previewing a theme.
+ * FIXME: This is copied from Gutenberg; we should be creating a selector for the `core/edit-site` store.
+ * @see https://github.com/WordPress/gutenberg/blob/053c8f733c85d80c891fa308b071b9a18e5194e9/packages/edit-site/src/utils/is-previewing-theme.js#L6
+ * @returns {string|null} currentlyPreviewingTheme
+ */
+function currentlyPreviewingTheme() {
+	if ( isPreviewingTheme() ) {
+		return getQueryArg( window.location.href, 'wp_theme_preview' );
+	}
+	return null;
+}
+
+/**
+ * Sometimes Gutenberg doesn't allow you to re-register the module and throws an error.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+let unlock: ( object: any ) => any | undefined;
+try {
+	unlock = __dangerousOptInToUnstableAPIsOnlyForCoreModules(
+		'I know using unstable features means my plugin or theme will inevitably break on the next WordPress release.',
+		'@wordpress/edit-site'
+	).unlock;
+} catch ( error ) {
+	// eslint-disable-next-line no-console
+	console.error( 'Error: Unable to get the unlock api. Reason: %s', error );
+}
+
+const NOTICE_ID = 'wpcom-live-preview/notice';
+
+/**
+ * This is an interim solution to clarify to users that they are currently live previewing a theme.
+ * And this should be moved to jetpack-mu-wpcom.
+ * @see https://github.com/Automattic/wp-calypso/issues/82218
+ */
+const LivePreviewNotice = () => {
+	const { createWarningNotice } = useDispatch( 'core/notices' );
+	const siteEditorStore = useSelect( ( select ) => select( 'core/edit-site' ), [] );
+	const { dashboardLink, previewingTheme } = useSelect(
+		( select ) => {
+			if ( ! siteEditorStore ) {
+				return {
+					previewingTheme: undefined,
+					dashboardLink: undefined,
+				};
+			}
+
+			const { getSettings } = unlock( siteEditorStore );
+			const themeSlug = currentlyPreviewingTheme();
+			// eslint-disable-next-line @typescript-eslint/no-explicit-any
+			const theme = ( select( 'core' ) as any ).getTheme( themeSlug );
+			return {
+				previewingTheme: theme?.name?.rendered || themeSlug,
+				dashboardLink: getSettings().__experimentalDashboardLink,
```

Comment:
I have a question, why not simply do the following:

```
dashboardLink: `/themes/${siteSlug}`,
```

because it's literally the back link?

This way we don't need to unlock the store.

Reply:
I wanted to make it consistent with Gutenberg, but it's nice that we can remove `unlock`. I'll address it!

Reply:
Sorry scratch this; I just realized that the current top-level URL is not `wordpress.com`, but rather the current site, so it won't work....